File: frontend/src/services/feedback.service.ts
import axiosClient from "@/lib/axios-client";
import { Feedback, FeedbackRequest, IssueType, FeedbackStatus } from "../features/feedbacks/types/feedback.type";

// Let's redefine PaginatedResponse here to avoid circular dependency or incorrect imports
export interface PaginatedFeedbackResponse {
  data: Feedback[];
  currentPage: number;
  totalPages: number;
  totalItems: number;
  hasNext: boolean;
  hasPrevious: boolean;
}

export const feedbackService = {
  getFeedbacks: async (
    page: number = 1,
    size: number = 10,
    issueType?: IssueType,
    status?: FeedbackStatus
  ): Promise<PaginatedFeedbackResponse> => {
    const params = new URLSearchParams({
      page: page.toString(),
      size: size.toString(),
    });
    
    if (issueType && issueType !== "ALL" as any) params.append("issue_type", issueType);
    if (status && status !== "ALL" as any) params.append("status", status);

    const response = await axiosClient.get(`/feedbacks?${params.toString()}`);
    return response.data;
  },

  getFeedbackById: async (id: number): Promise<Feedback> => {
    const response = await axiosClient.get(`/feedbacks/${id}`);
    return response.data;
  },

  createFeedback: async (request: FeedbackRequest): Promise<Feedback> => {
    const response = await axiosClient.post("/feedbacks", request);
    return response.data;
  },

  updateFeedback: async (id: number, request: FeedbackRequest): Promise<Feedback> => {
    const response = await axiosClient.put(`/feedbacks/${id}`, request);
    return response.data;
  },
};

File: backend/src/main/java/vn/edu/ut/pbms/service/FeedbackService.java
package vn.edu.ut.pbms.service;

import org.springframework.data.domain.Page;
import vn.edu.ut.pbms.constant.FeedbackStatus;
import vn.edu.ut.pbms.constant.IssueType;
import vn.edu.ut.pbms.dto.request.FeedbackRequest;
import vn.edu.ut.pbms.dto.response.FeedbackResponse;

public interface FeedbackService {
    Page<FeedbackResponse> getAllFeedbacks(IssueType issueType, FeedbackStatus status, int page, int size);
    FeedbackResponse getFeedbackById(Long id);
    FeedbackResponse createFeedback(FeedbackRequest request);
    FeedbackResponse updateFeedback(Long id, FeedbackRequest request);
}
